e2e/tests/rich-text/rich-text-perfomance.test.js

```diff
@@ -0,0 +1,127 @@
+const assert = require('assert');
+const { createRandomIntWithSeed } = require('../helpers');
+
+Feature('Richtext perfomance');
+
+Before(({ LabelStudio }) => {
+  LabelStudio.setFeatureFlags({
+    ff_front_1170_outliner_030222_short: true,
+    fflag_feat_front_lsdv_4620_richtext_opimization_060423_short: true,
+  });
+});
+
+// Generate text and results generator
+const SYMBOLS = 'abcdefghijklmnopqrstuvwxyz';
+const NEWLINE = '/n';
+const SPACE = ' ';
+const MIN_WORD_SIZE = 2;
+const MAX_WORD_SIZE = 23;
+
+function generateAnnotationParams(sybolsNum, regionsNum, seed = 42) {
+  const genWord = (len) => {
+    const start = length % SYMBOLS.length;
+    const end = start + len;
+    const s = SYMBOLS.repeat(Math.ceil(end / SYMBOLS.length));
+
+    return s.substring(start, end);
+  };
+  const words = [];
+  const randomInt = createRandomIntWithSeed(seed);
+  let length = 0;
+  let paragraphWords = randomInt(0, 150);
+
+  let currentWordLength;
+
+  while ((currentWordLength = Math.min(randomInt(MIN_WORD_SIZE, MAX_WORD_SIZE), sybolsNum - length)) > 0) {
+    words.push(genWord(currentWordLength));
```

Comment:
Is this performance at all based on line length, not just the number of segments? I ask only because this test even though deterministically random, won't necessarily produce large line lengths.

Reply:
As I see this problem it's more about number of regions and the whole size of content. In the test I just tried to emulate natural variability of results, just in case.

Reply:
That's fair, I think this does a good job of representing that.